python/lsst/daf/butler/_butler.py

```diff
@@ -1705,10 +1705,10 @@ def doImport(importStream: TextIO) -> None:
         else:
             doImport(filename)
 
-    @transactional
```

Comment:
What is the reason you cannot keep `@transactional` here?

Reply:
Because we register dataset types and we explicitly disallow that inside an existing transaction.